packages/server/dataloader/azureDevOpsLoaders.ts

```diff
@@ -0,0 +1,495 @@
+          const now = new Date()
+          const inAMinute = Math.floor((now.getTime() + 60000) / 1000)
+          if (!decodedToken || decodedToken.exp < inAMinute) {
+            // console.log(`calling AzureDevOpsServerManager.refresh`)
+            // const oauthRes = await AzureDevOpsServerManager.refresh(refreshToken)
+            // console.log(`oauthRes: ${JSON.stringify(oauthRes)}`)
+            if (!refreshToken || !accessTokenSecret) {
+              return null
+            }
+
+            const provider = await parent.get('integrationProviders').loadNonNull(providerId)
+
+            if (!provider) {
+              return null
+            }
+
+            const manager = new AzureDevOpsServerManager(azureDevOpsAuthToRefresh, provider as IntegrationProviderAzureDevOps)
+
+            const oauthRes = await manager.refresh(refreshToken)
+            if (oauthRes instanceof Error) {
+              //sendToSentry(oautRes)
+              return null
+            }
+            const {accessToken, refreshToken: newRefreshToken} = oauthRes
+            console.log(`accessToken: ${accessToken} | newRefreshToken: ${newRefreshToken}`)
+            const updatedRefreshToken = !!newRefreshToken ? newRefreshToken : refreshToken
+            // const updatedSameAccountAzureDevOpsAuths = userAzureDevOpsAuths
+            //   .filter((auth) => auth.accountId === azureDevOpsAuthToRefresh.accountId)
+            //   .map((auth) => ({
+            //     ...auth,
+            //     accessToken,
+            //     refreshToken: updatedRefreshToken
+            //   }))
+            // console.log(updatedSameAccountAzureDevOpsAuths)
+            // await upsertAzureDevOpsAuths(updatedSameAccountAzureDevOpsAuths)
+            const newAzureDevOpsAuth = {
+              ...azureDevOpsAuthToRefresh,
+              accessToken,
+              refreshToken: updatedRefreshToken
+            }
+            await upsertTeamMemberIntegrationAuth(newAzureDevOpsAuth)
+            console.log(`newAzureDevOpsAuth: ${newAzureDevOpsAuth}`)
+            return newAzureDevOpsAuth
+          }
+          console.log(`azureDevOpsAuthToRefresh: ${azureDevOpsAuthToRefresh}`)
+          return azureDevOpsAuthToRefresh
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : null))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.userId}:${key.teamId}`
+    }
+  )
+}
+
+export const azureDevOpsAllWorkItems = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, WorkItem[] | undefined, string> => {
+  return new DataLoader<TeamUserKey, WorkItem[] | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const returnWorkItems = [] as WorkItem[]
+          // const auth = await parent.get('freshAzureDevOpsAuth').load({teamId, userId})
```

Comment:
-1 getting 'freshAzureDevOpsAuth' here was the right call. Same for all other functions here. You can also think about modifying its return type to include the provider since you always need it.